### DISCUSSION TITLE
Connecting VSCode Desktop (Locally) to Azure Studio Compute Instance through Azure Machine Learning Extension

### DISCUSSION THREAD

**Discussion by aiplayer93** [microsoft/vscode-discussions#1247] [Extension Development QnA] [answered]
Hello guys,,

I have been combing the internet for this for a long long time, so your help is very very much appreciated,

I am trying to connect to an Azure Studio Compute Instance through VSCode Desktop and its associated Azure Machine Learning VSCode extension.

1. Context: While developing in Azure Studio ML Compute instance, I would need to develop in VSCode Desktop locally (VSCode Web in the Azure Studio I have is freezing and the Desktop version link in Azure Studio does not open my VSCode with administrative privileges - and for the work I do I need to develop locally with elevated privileges).

I have tried to connect from VSCode Desktop (local VSCode) to an Azure ML Compute Instance through the Azure Machine Learning Extension (and the associated Remote) last week. It did work through VS Code Azure extension and then suddenly broke a few days later. The error message varies depending on the vs code extension that is being used.

Usually however, the error is either that

the Azure Extension is not installed on the remote or
that the remote listening is not on
that it failed to connect to the remote extension host server (Error: Cannot read properties of undefined (reading 'sessionId')
cannot read propoerties of undefined (reading 'sessionId') Source Azure Machine Learning - Remote

![image](https://github.com/microsoft/vscode-discussions/assets/113632605/9ab69a50-9ca3-4199-8578-3834125053d2)

![image](https://github.com/microsoft/vscode-discussions/assets/113632605/f5943390-91d6-4e12-b796-9bd1b27109de)

![image](https://github.com/microsoft/vscode-discussions/assets/113632605/552abe58-280f-467b-ab7a-a80528f6acba)

2. The question: So I have managed to make it work before by tweaking the different version number of Azure Machine Learning and Azure Machine Learning Remote at 0.28.0 and 0.32.0 respectively (the most up to date version gave errors and I tweaked on the different versions). The VSCode version I have now is 1.90.2. Please note I have tried to create a new workspace in Azure Studio and new computes and also reinstalled VSCode to the current 1.90.2 version.

Is this a matter of version compatability and what would be the best combinations of version to make it work? What would a quick fix be for this? Should I consider the ssh route for this instead of the extension (note our compute instance is not allowed for now to have a public IP due to the vnet restrictions that are applied on the workspace).

**Comment by gjsjohnmurray** [comment, marked as answer]
This seems to be a repost of #1245 but in a different discussion category. Yet I still don't think you are **developing** an extension for VS Code. Please review the READ FIRST post (#5).
